File: samples/hello.go
// TruePink — Go
package main

import (
	"fmt"
	"strings"
)

const MaxLives = 9

type Cat struct {
	Name  string
	Lives int
}

func (c *Cat) Meow(loud bool) string {
	msg := fmt.Sprintf("%s says meow", c.Name)
	if loud {
		return strings.ToUpper(msg)
	}
	return msg
}

func main() {
	cat := &Cat{Name: "Mochi", Lives: MaxLives}
	nums := []int{1, 2, 3, 0xFF}
	for i, n := range nums {
		if n%2 == 0 {
			fmt.Println(i, cat.Meow(true))
		}
	}
}

File: samples/hello.rs
// TruePink — Rust
use std::collections::HashMap;

#[derive(Debug)]
struct Cat {
    name: String,
    lives: u32,
}

impl Cat {
    fn new(name: &str) -> Self {
        Cat { name: name.to_string(), lives: 9 }
    }

    fn meow(&self, loud: bool) -> String {
        let msg = format!("{} says meow", self.name);
        if loud { msg.to_uppercase() } else { msg }
    }
}

const MAX: u32 = 100;

fn main() {
    let mut map: HashMap<&str, i32> = HashMap::new();
    map.insert("pink", 0xFF);
    let cat = Cat::new("Mochi");
    for i in 0..MAX {
        if i % 2 == 0 {
            println!("{}", cat.meow(true));
        }
    }
}
